clients/ts/signalr/src/Polyfills.ts

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+// Not exported from index
+
+export interface EventSourceConstructor {
```

Comment:
Why is this so weird? Is this a ts/js thing? Why not just make it an interface?

Reply:
So this is just describing the browser's EventSource and WS as a typescript interface?

Reply:
Kinda, it's describing the **constructor function** as a TypeScript Type so it can be used as the type of a property. In TypeScript an interface can represent anything, including a function. This is the syntax for making an interface that describes a constructor function (and in the case of `WebSocketConstructor`, it also includes static properties).

The TypeScript built-in DOM typings already include interfaces for `WebSocket` and `EventSource` **instances** but [the way they represent the constructor function itself](https://github.com/Microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L4662) isn't usable as a type, they just declare it as a global variable with an in-line type. This is basically just taking that in-line type and making it into a named type so the global `WebSocket` constructor function can be passed in.